python/pyarrow/tests/test_array.py

```diff
@@ -3579,6 +3579,18 @@ def test_run_end_encoded_from_buffers():
                                            1, offset, children)
 
 
+def test_run_end_encoded_from_array_with_type():
+    arr = [1, 2, 2, 3, 3, 3]
```

Comment:
This will also have to work for other kinds of inputs (eg np.ndarray instead of a python list)